x-pack/plugins/cloud_security_posture/server/lib/check_index_status.ts

```diff
@@ -6,25 +6,33 @@
  */
 
 import { ElasticsearchClient, type Logger } from '@kbn/core/server';
-import { LATEST_FINDINGS_INDEX_DEFAULT_NS, FINDINGS_INDEX_PATTERN } from '../../common/constants';
+import { IndexStatus } from '../../common/types';
 
-export const checkForFindings = async (
+export const checkIndexStatus = async (
   esClient: ElasticsearchClient,
-  latestIndex: boolean,
+  index: string,
   logger: Logger
-): Promise<boolean> => {
+): Promise<IndexStatus> => {
   try {
     const queryResult = await esClient.search({
-      index: latestIndex ? LATEST_FINDINGS_INDEX_DEFAULT_NS : FINDINGS_INDEX_PATTERN,
+      index,
       query: {
         match_all: {},
       },
       size: 1,
     });
 
-    return !!queryResult.hits.hits.length;
+    if (queryResult.hits.hits.length) {
+      return 'exists';
+    }
+
+    return 'empty';
   } catch (e) {
     logger.debug(e);
-    return false;
+    if (e?.meta?.body?.error?.type === 'security_exception') {
+      return 'unprivileged';
+    }
+
+    return 'empty';
```

Comment:
`empty` is returned twice, in different cases. are we sure both cases are the same? if not, consider adding `none-exists` instead

Reply:
so we can't tell `none-exists`. It differs if querying an index or index-pattern. And I'm guessing there are more possible edge cases here